source/Halibut/Transport/Protocol/RequestMessageWithCancellationToken.cs

```diff
@@ -0,0 +1,16 @@
+using System.Threading;
+
+namespace Halibut.Transport.Protocol
+{
+    public class RequestMessageWithCancellationToken
```

Comment:
This contains the request message and a cancellation token that the dequeuing thread can observe to allow us to cancel across the queue.